### PR TITLE
Oscar/seq drydep set h coeff

### PR DESCRIPTION
**Pull request by odiazib**
We've ported the[ set_hcoeff_vector](https://github.com/eagles-project/e3sm_mam4_refactor/blob/be423b9ce96c7d842a462fca9e671d4b5b710c87/driver-mct/shr/seq_drydep_mod.F90#L1176) routine.

This routine utilizes dheff (the effective Henry's Law coefficient), defined [here](https://github.com/eagles-project/e3sm_mam4_refactor/blob/be423b9ce96c7d842a462fca9e671d4b5b710c87/driver-mct/shr/seq_drydep_mod.F90#L631).

Originally, dheff includes definitions for 192 species. In our ported version, we have customized this table to include only 3 gas species currently used in our base case simulation: `H2O2`, `H2SO4`, and `SO2`.

### CODE REVIEW

**Review by singhbalwinder — APPROVED** [review]

**Comment by singhbalwinder** [comment]
Looks great! I just left some minor comments.

File: src/mam4xx/seq_drydep.hpp [hunk 1]
```diff
@@ -55,13 +59,83 @@ struct Data {
...
+      1.36e+00, 3100., 1.30e-02, 1960.,  6.6e-08, 1500. // SO2
+  };
+  constexpr int mapping[n_drydep] = {1, 2, 3};
+  // BAD CONSTANT
+  constexpr Real ph = 1.e-5; // measure of the acidity (dimensionless)
+
+  const Real t0 = 298.0;        // Standard Temperature
+  const Real ph_inv = 1.0 / ph; // Inverse of PH
+
+  const Real wrk = (t0 - sfc_temp) / (t0 * sfc_temp);
```
[the remark this answers is not in the document]
Comment (odiazib) [reply src/mam4xx/seq_drydep.hpp:97]: `t0` is 298, and `sfc_temp` (surface temperature) is very unlikely to be zero. Thus, I do not think we will encounter a division by zero. In addition, if `sfc_temp` were to be zero, an error would be produced before any threads reach this point.

**Review by mingxuanwupnnl — APPROVED** [review]
@odiazib Thank you for porting this subroutine. It looks good to me. I don't have other comments.

File: src/mam4xx/seq_drydep.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (singhbalwinder) [reply src/mam4xx/seq_drydep.hpp:97]: Yeah, that is a good point. I also think it is unlikely to get into divide by zero error here.

**Pull request merged**